### PR TITLE
fix: onRowSelectionChange is called twice on every click on the row #373

### PR DESCRIPTION
**Pull request by agmitron**
[DataTable onRowSelectionChange event doesn't work correct](https://app.zenhub.com/workspaces/jmix-frontend-issues-5fe0992a4308f20011f1c913/issues/haulmont/jmix-frontend/373) #373

### CODE REVIEW

File: packages/react-ide-toolbox/src/previews/component-preview.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ interface Props {
   exact?: boolean;
 }
 
-export const ComponentPreview: React.FC<Props> = ({ path, children, setToolsPropsToEdit, exact = false, propsEditInfo }: Props) => {
+export const ComponentPreview: React.FC<Props> = ({ path, children, setToolsPropsToEdit, exact = true, propsEditInfo }: Props) => {
```
Comment (vyacheslav-pushkin) [inline packages/react-ide-toolbox/src/previews/component-preview.tsx:13]: What is the purpose of this change?

File: packages/jmix-react-ui/src/ui/table/DataTable.test.tsx [hunk 2]
```diff
@@ -1,15 +1,19 @@
 import '@testing-library/jest-dom';
-import {render, screen} from '@testing-library/react'
+import {create, ReactTestInstance} from 'react-test-renderer';
 import React from 'react';
 import {DataTable} from "./DataTable";
 import {IntlProvider} from "react-intl";
 import { Provider } from 'mobx-react';
 import { MainStore } from '@haulmont/jmix-react-core';
 
+// import {Simulate} from 'react-dom/test-utils';
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-ui/src/ui/table/DataTable.test.tsx:9]: We should not commit commented-out code.

File: packages/jmix-react-ui/src/ui/table/DataTable.test.tsx [hunk 3]
```diff
@@ -24,29 +28,78 @@ window.matchMedia = () => ({
...
+    rows[0].props.onClick(new Event('click'))
+    const newSelectedRowKeys = dataTableComponent.instance.selectedRowKeys;
+    expect(prevSelectedRowKeys).not.toEqual(newSelectedRowKeys);
+    expect(newSelectedRowKeys).toHaveLength(1);
+
+    await wait(1000);
+
+    // ??? Why it isn't working? TODO: think a little about it later
+    // expect(onRowSelectionChangeSpy).toBeCalled();
+    // expect(onRowSelectionChangeSpy).toHaveBeenCalledTimes(1);
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-ui/src/ui/table/DataTable.test.tsx:94]: Let's remove the call to `wait` since we don't assert anything afterwards.

Comment (vyacheslav-pushkin) [reply packages/jmix-react-ui/src/ui/table/DataTable.test.tsx:94]: Also, using timeouts like this can lead to fragile tests (it works locally, then *sometimes* it fails on CI, etc.)

`@testing-library/react` has some [async utilities](https://testing-library.com/docs/dom-testing-library/api-async/) that are a more robust way to wait for something.

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: packages/react-ide-toolbox/src/previews/component-preview.tsx (hunk 1 above)
Comment (agmitron) [reply packages/react-ide-toolbox/src/previews/component-preview.tsx:13]: It was rebased from master

Comment (vyacheslav-pushkin) [reply packages/react-ide-toolbox/src/previews/component-preview.tsx:13]: This change is not on master, otherwise it wouldn't be visible in the PR.

Comment (agmitron) [reply packages/react-ide-toolbox/src/previews/component-preview.tsx:13]: https://github.com/Haulmont/jmix-frontend/commit/449e259caf99b8da8dfee943fde5a9a629f016be

**Review by vyacheslav-pushkin — APPROVED** [review]
